### PR TITLE
BC-3341 -  FE - Display downloadable file in file element

### PR DESCRIPTION
**Pull request by bergatco**
# Short Description

Display selected file in file element.

## Links to Ticket and related Pull-Requests

https://ticketsystem.dbildungscloud.de/browse/BC-3341

Based on work done in :
https://ticketsystem.dbildungscloud.de/browse/BC-3296
https://github.com/hpi-schul-cloud/nuxt-client/pull/2542

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

File: src/components/feature-board/content-elements/FileContentElementDisplay.vue [hunk 1]
```diff
@@ -15,6 +24,79 @@ export default defineComponent({
 			type: String,
 			required: true,
 		},
+		fileRecord: {
+			type: Object as () => FileRecordResponse,
```
Comment (bischofmax) [inline src/components/feature-board/content-elements/FileContentElementDisplay.vue:28]:
```suggestion
			type: Object as PropType<FileRecordResponse>,
```

Comment (bergatco) [reply src/components/feature-board/content-elements/FileContentElementDisplay.vue]: Fixed! Also for `FileContentElementEdit`.

File: src/components/feature-board/content-elements/FileContentElementDisplay.vue [hunk 2]
```diff
@@ -1,12 +1,21 @@
...
-	<div>
-		<div>
+	<div class="file-element margin">
+		<div class="file-card">
+			<v-icon x-large>{{ mdiFileDocumentOutline }}</v-icon>
+			<a class="file-card-label truncate" v-bind:href="fileRecord.url" download>
+				{{ fileRecord.name }}
+			</a>
+		</div>
+		<div class="file-info">
```
Comment (SevenWaysDP) [inline src/components/feature-board/content-elements/FileContentElementDisplay.vue:9]: Bitte lass uns passende vuetify elemente finden

File: src/components/feature-board/board/BoardColumn.vue [hunk 3]
```diff
@@ -10,6 +10,7 @@
 			@delete:column="onColumnDelete"
 			@move:column-keyboard="onMoveColumnKeyboard"
 			@update:title="onUpdateTitle"
+			class="pl-2"
```
Comment (SevenWaysDP) [inline src/components/feature-board/board/BoardColumn.vue:13]: warum müssen wir so globalw Änderungen machen um unserer Element richtig darzustellen

**Review by SevenWaysDP — CHANGES_REQUESTED** [review]

File: src/components/feature-board/board/BoardColumn.vue (hunk 3 above)
Comment (bischofmax) [reply src/components/feature-board/board/BoardColumn.vue:13]: @bergatco Ich glaube das wird nur von GitHub wieder falsch dargestellt oder? Kommen die Änderungen von Dir?

File: src/components/feature-board/content-elements/FileContentElementDisplay.unit.ts [hunk 4]
```diff
@@ -3,17 +3,29 @@ import { shallowMount } from "@vue/test-utils";
 import FileContentElementDisplay from "./FileContentElementDisplay.vue";
 
 describe("FileContentElementDisplay", () => {
+	const setupProps = () => ({
+		caption: "Test Caption",
+		fileRecord: {
```
[the remark this answers is not in the document]
Comment (bergatco) [reply src/components/feature-board/content-elements/FileContentElementDisplay.unit.ts]: Fixed!

File: src/components/feature-board/types/DragAndDrop.ts [hunk 5]
```diff
@@ -23,7 +23,7 @@ export type DragAndDropKey =
 	| "Enter";
 
 export const cardDropPlaceholderOptions = {
-	className: "rounded-sm grey lighten-1 my-3",
+	className: "rounded-sm grey lighten-1 my-3 mx-3",
```
[the remark this answers is not in the document]
Comment (bergatco) [reply src/components/feature-board/types/DragAndDrop.ts:26]: Ignored as not part of this PR! Changes seem to come from the `main` branch.

**Review by SevenWaysDP — APPROVED** [review]

File: src/components/feature-board/types/DragAndDrop.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (bergatco) [reply src/components/feature-board/types/DragAndDrop.ts:26]: Ignored as not part of this PR! Changes seem to come from the `main` branch.

File: src/components/feature-board/card/CardHost.vue [hunk 6]
```diff
@@ -204,3 +204,15 @@ export default defineComponent({
...
+.v-card:focus::before {
+	opacity: 0;
+}
+
+.v-card:focus,
+.v-card:focus-within {
+	outline: 2px solid var(--v-secondary-base);
+	outline-offset: 0;
+}
+</style>
```
[the remark this answers is not in the document]
Comment (bergatco) [reply src/components/feature-board/card/CardHost.vue:218]: Ignored as not part of this PR! Changes seem to come from the `main` branch.

File: src/components/feature-board/content-elements/FileContentElementDisplay.vue (hunk 2 above)
Comment (bergatco) [reply src/components/feature-board/content-elements/FileContentElementDisplay.vue]: Fixed!

File: src/components/feature-board/board/BoardColumn.vue (hunk 3 above)
Comment (bergatco) [reply src/components/feature-board/board/BoardColumn.vue:13]: Ignored as not part of this PR! Changes seem to come from the `main` branch.

**Review by Loki-Afro — APPROVED** [review]

**Pull request merged**